src/runtime-rs/crates/persist/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+pub mod hypervisor;
+pub mod network;
+pub mod sandbox;
+use anyhow::{anyhow, Ok, Result};
+use serde::de;
+use std::{fs::File, io::BufReader};
+pub const KATA_PATH: &str = "/run/kata";
+pub const PERSIST_FILE: &str = "persist.json";
+use async_trait::async_trait;
+
+#[async_trait]
+pub trait Persist
+where
+    Self: Sized,
+{
+    /// The type of the object representing the state of the component.
+    type State;
+    /// The type of the object holding the constructor arguments.
+    type ConstructorArgs;
+
+    /// Returns the current state of the component.
+    async fn save(&self) -> Result<Self::State>;
+    /// Constructs a component from a specified state.
+    async fn restore(constructor_args: Self::ConstructorArgs, state: &Self::State) -> Result<Self>;
+}
+
+pub fn to_disk<T: serde::Serialize>(value: &T, sid: &str) -> Result<()> {
+    let sandbox_file = [KATA_PATH, sid, PERSIST_FILE].join("/");
+    let f = File::create(sandbox_file)?;
+    let j = serde_json::to_value(value)?;
+    serde_json::to_writer_pretty(f, &j)?;
+    Ok(())
+}
+
+pub fn from_disk<T>(sid: &str) -> Result<T>
+where
+    T: de::DeserializeOwned,
+{
+    let sandbox_file = [KATA_PATH, sid, PERSIST_FILE].join("/");
+    let file = File::open(sandbox_file).unwrap();
```

Comment:
Same comment as above. This function could potentially read an arbitrary file given that `sid` is not being sanity checked:

If `sid` was `../../tmp/malicious-hacker-directory/`, file `/tmp/malicious-hacker-directory/persist.json` could be opened.

Reply:
fixed